src/os/bluestore/BlueStore.cc

```diff
@@ -6541,7 +6550,14 @@ int BlueStore::migrate_to_new_bluefs_device(const set<int>& devs_source,
   bluefs->add_block_extent(
     id, reserved, bluefs->get_block_device_size(id) - reserved);
 
-  r = bluefs->device_migrate_to_new(cct, devs_source, id);
+  if (!link_db.empty() && id != BlueFS::BDEV_NEWDB) {
```

Comment:
This part probably lacks db creation case, i.e. when link_db.empty() && id == NEWDB.
Then we starts having dedicated db device...

Reply:
Reworked.